cache_flat_mutation_reader.hh

```diff
@@ -642,10 +657,38 @@ void cache_flat_mutation_reader::copy_from_cache_to_buffer() {
     }, _read_context.is_reversed()) == stop_iteration::no) {
         return;
     }
-    // We add the row to the buffer even when it's full.
-    // This simplifies the code. For more info see #3139.
+
     if (_next_row_in_range) {
-        add_to_buffer(_next_row);
```

Comment:
The patch shouldn't close https://github.com/scylladb/scylladb/issues/6033 because it doesn't deal with all tombstones. It doesn't expire range tombstones, for example.

I don't think we should try to handle range tombstones before https://github.com/scylladb/scylladb/pull/12048 is merged, which makes significant changes to the structure of cache with regards to range tombstones and MVCC.

Reply:
I see. Do you think #12048 is expected to be merged soon?

Reply:
Yes. You can rebase on top of it as Avi suggested.